src/convertToObject.js

```diff
@@ -12,7 +12,20 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  let array = sourceString.split(';');
+
+  const result = {};
+
+  array = array.map(item => (item.replace('"').trim()));
+
+  while (array.indexOf('') !== -1) {
+    array.splice(array.indexOf(''), 1);
+  }
```

Comment:
Here you are overcomplicating things. As a result you will have three cycles inside one another.
I would suggest using filter to simply filter out empty array elements.